src/components/entity/ha-state-label-badge.ts

```diff
@@ -10,21 +10,43 @@ import {
 } from "lit";
 import { customElement, property, state } from "lit/decorators";
 import { classMap } from "lit/directives/class-map";
+import { arrayLiteralIncludes } from "../../common/array/literal-includes";
 import secondsToDuration from "../../common/datetime/seconds_to_duration";
 import { computeStateDisplay } from "../../common/entity/compute_state_display";
 import { computeStateDomain } from "../../common/entity/compute_state_domain";
 import { computeStateName } from "../../common/entity/compute_state_name";
+import { FIXED_DOMAIN_STATES } from "../../common/entity/get_states";
 import {
   formatNumber,
   getNumberFormatOptions,
   isNumericState,
 } from "../../common/number/format_number";
-import { UNAVAILABLE, UNKNOWN } from "../../data/entity";
+import { isUnavailableState, UNAVAILABLE, UNKNOWN } from "../../data/entity";
 import { timerTimeRemaining } from "../../data/timer";
 import { HomeAssistant } from "../../types";
 import "../ha-label-badge";
 import "../ha-state-icon";
 
+// Define the domains whose states have special truncated strings
+const TRUNCATED_DOMAINS = [
+  "alarm_control_panel",
+  "device_tracker",
+  "person",
+] as const satisfies ReadonlyArray<keyof typeof FIXED_DOMAIN_STATES>;
+
+type TruncatedDomain = typeof TRUNCATED_DOMAINS[number];
+type TruncatedKey = {
+  [T in TruncatedDomain]: `${T}.${typeof FIXED_DOMAIN_STATES[T][number]}`;
+}[TruncatedDomain];
+
+const isTruncatedKey = (key: `${string}.${string}`): key is TruncatedKey => {
+  const [domainKey, stateKey] = key.split(".", 2);
+  return (
+    arrayLiteralIncludes(TRUNCATED_DOMAINS)(domainKey) &&
+    arrayLiteralIncludes(FIXED_DOMAIN_STATES[domainKey])(stateKey)
+  );
+};
```

Comment:
I suppose that would be valid as well (without checking with TS), but not sure how much more performant.  As is, you'd trade the array split for guaranteeing 2 include checks instead of mostly 1, so I'd rearrange the return order at least.

Reply:
Done.